common/engine/keyboardprocessor/tests/unit/kmx/meson.build

```diff
@@ -48,24 +48,29 @@ kmns = [
 #  '033 - caps alwyas off' # TODO: support capsononly, capsalwaysoff, shiftfreescaps
 ]
 
-prog_kmcomp = find_program('kmcomp')
+prog_kmcomp = find_program('kmcomp', required: false)
 
-# todo: if kmcomp is not found, use the .kmx in the source folder
-# therefore make kmcomp compile to the source folder and add to repo?
+if prog_kmcomp.found()
+  foreach kmn : kmns 
+    kmn_test = custom_target(
+      kmn + '.kmn',
+      output: kmn + '.kmx',
+      input: kmn + '.kmn',
+      command : [prog_kmcomp, '-s', '@INPUT@']
+    )
+    
+    #kmn_source = configure_file(
+      #input: 'kmn/' + kmn + '.kmn',
+      #output: kmn + '.kmn',
+      #copy: true
+    #)
 
-foreach kmn : kmns 
-  kmn_test = custom_target(
-    kmn + '.kmn',
-    output: kmn + '.kmx',
-    input: 'kmn/' + kmn + '.kmn',
-    command : [prog_kmcomp, '-s', '@INPUT@', '@OUTPUT@']
-  )
-  
-  kmn_source = configure_file(
-    input: 'kmn/' + kmn + '.kmn',
-    output: kmn + '.kmn',
-    copy: true
-  )
+    test(kmn, kmx, depends: [kmn_test], args: [kmn])
```

Comment:
this isn't a test, I don't think it is needed when the tests are generated at the bottom whether there is kmcomp or not

Reply:
Yes, correct.